lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -121,12 +121,13 @@ IRGenDebugInfo::IRGenDebugInfo(const IRGenOptions &Opts,
   }
 
   unsigned Lang = llvm::dwarf::DW_LANG_Swift;
-  std::string Producer = version::getSwiftFullVersion();
+  std::string Producer = version::getSwiftFullVersion(
+    Opts.EffectiveLanguageVersion);
   bool IsOptimized = Opts.Optimize;
   StringRef Flags = Opts.DWARFDebugFlags;
-  unsigned Major, Minor;
-  std::tie(Major, Minor) = version::getSwiftNumericVersion();
-  unsigned MajorRuntimeVersion = Major;
+
+  assert(!Opts.EffectiveLanguageVersion.empty());
```

Comment:
We are using the DW_AT_APPLE_runtime_version in the DW_LANG_Swift compile unit to encode the Swift runtime version (actually the swift compiler version because they are thus far identical, cf. rdar://problem/18729762).

The Objective-C runtime version can be found in the second, DW_LANG_ObjC compile unit that is also part of each object file generated by swiftc.